app/api/server/v1/voip/omnichannel.ts

```diff
@@ -0,0 +1,134 @@
+import { Match, check } from 'meteor/check';
+
+import { API } from '../../api';
+import { Users } from '../../../../models/server/raw/index';
+import { hasPermission } from '../../../../authorization/server/index';
+import { LivechatVoip } from '../../../../../server/sdk';
+
+API.v1.addRoute('omnichannel/agent/extension', { authRequired: true }, {
+	// Get the extensions associated with the agent passed as request params.
+	async get() {
+		if (!hasPermission(this.userId, 'view-agent-extension-association')) {
+			return API.v1.unauthorized();
+		}
+		check(this.requestParams(), Match.ObjectIncluding({
+			username: String,
+		}));
+		const { username } = this.requestParams();
+		const user = await Users.findOneByAgentUsername(username, {
+			projection: { _id: 1 },
+		});
+		if (!user) {
+			return API.v1.notFound('User not found');
+		}
+		const extension = await Users.getVoipExtensionByUserId(user._id, {
+			projection: {
+				_id: 1,
+				username: 1,
+				extension: 1,
+			},
+		}).toArray();
+		if (!extension) {
+			return API.v1.notFound('Extension not found');
+		}
+		return API.v1.success({ extension });
+	},
+
+	// Create agent-extension association.
+	async post() {
```

Comment:
Should we validate extension is a number?

Reply:
Well, SIP extensions can be non numeric texts. https://en.wikipedia.org/wiki/SIP_URI_scheme#:~:text=A%20SIP%20address%20is%20written,voip%2Dprovider.example.net&text=As%20a%20SIP%20address%20is,may%20contain%20non%2Dnumeric%20characters.

Right now I would keep it simple. But once we have people using it and we discover different formats people use, we can add some validations. 

Furthermore, as Diego pointed out yesterday, while doing this association, we may have to check on asterisk if the extension really exists. So I would prefer keeping it simple at this point of time.